spyder/utils/introspection/utils.py

```diff
@@ -210,6 +210,45 @@ def get_keywords(lexer):
                 continue
     return keywords
 
+def get_words_file(file_path=None, content=None, extension=None):
```

Comment:
This name is not appropriate as it assumes file.... `get_words` is enough

The code has repeated code

```python

def get_words_file(file_path=None, content=None, extension=None):
    """
    Extract all words from source code string.

    Extract the list of words that contains the file in the editor,
    to carry out the inline completion similar to VSCode.
    """
    if (file_path is None and (content is None or extension is None) or
                    file_path and content and extension):
        error_msg = ('Must provide `file_path` or `content` and `extension`')
        raise Exception(error_msg)

    if file_path and content is None and extension is None:
        ext = os.path.splitext(file_path)[1]

        with open(file_path, 'r') as infile:
            content = infile.read()

    if extension in ['.css']:
        regex = re.compile(r'([^a-zA-Z-])')
    elif extension in ['.R', '.c', 'md', '.cpp, java', '.py']:
        regex = re.compile(r'([^a-zA-Z_])')
    else:
        regex = re.compile(r'([^a-zA-Z])')

    words = sorted(set(regex.sub(r' ', content).split()))
    return words
```